src/setup.rs

```diff
@@ -66,18 +78,25 @@ pub(crate) fn initial_state(config: config::SpotifydConfig) -> main_loop::MainLo
     let autoplay = config.autoplay;
     let device_id = session_config.device_id.clone();
 
+    let default_volume_ctrl = if matches!(config.volume_controller, config::VolumeController::None)
+    {
+        VolumeCtrl::Fixed
+    } else {
+        VolumeCtrl::default()
+    };
+
     #[cfg(feature = "alsa_backend")]
     let volume_ctrl = if matches!(
         config.volume_controller,
         config::VolumeController::AlsaLinear
     ) {
         VolumeCtrl::Linear
     } else {
-        VolumeCtrl::default()
+        default_volume_ctrl
```

Comment:
If my above suggestion works, we should be able to merge these sections similarly as well.

Reply:
True!